cli/azd/pkg/infra/provisioning/terraform/terraform_provider.go

```diff
@@ -0,0 +1,595 @@
+package terraform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/infra"
+	. "github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/terraform"
+)
+
+// TerraformProvider exposes infrastructure provisioning using Azure Terraform templates
+type TerraformProvider struct {
+	env         *environment.Environment
+	projectPath string
+	options     Options
+	console     input.Console
+	cli         terraform.TerraformCli
+	azCli       azcli.AzCli
+}
+
+type TerraformDeploymentDetails struct {
+	ParameterFilePath  string
+	PlanFilePath       string
+	localStateFilePath string
+}
+
+// Name gets the name of the infra provider
+func (t *TerraformProvider) Name() string {
+	return "Terraform"
+}
+
+func (t *TerraformProvider) RequiredExternalTools() []tools.ExternalTool {
+	return []tools.ExternalTool{t.cli}
+}
+
+// NewTerraformProvider creates a new instance of a Terraform Infra provider
+func NewTerraformProvider(ctx context.Context, env *environment.Environment, projectPath string, infraOptions Options) *TerraformProvider {
+	terraformCli := terraform.GetTerraformCli(ctx)
+	console := input.GetConsole(ctx)
+	azCli := azcli.GetAzCli(ctx)
+
+	// Default to a module named "main" if not specified.
+	if strings.TrimSpace(infraOptions.Module) == "" {
```

Comment:
No need to change things here (and you're just following existing patterns), but long term I think we should try to centralize all of this logic which applies defaults to `azure.yaml` into the project file reader stuff.

Reply:
yes, could come in the next iteration